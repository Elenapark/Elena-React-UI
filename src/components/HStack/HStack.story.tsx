import React, { FC } from 'react';
import { Meta } from '@storybook/react';
import HStack from './HStack';
import { CustomStackProps } from 'components/Stack';

export default {
  title: 'HStack',
  component: HStack,
} as Meta<Omit<CustomStackProps, 'direction'>>;

export const Default: FC = () => {
  return (
    <HStack spacing="24px">
      <div style={{ background: 'pink' }}>Child1</div>
      <div style={{ background: 'pink' }}>Child2</div>
      <div style={{ background: 'pink' }}>Child3</div>
    </HStack>
  );
};
