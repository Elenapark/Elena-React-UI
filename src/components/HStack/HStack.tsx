import React from 'react';
import Stack from '../Stack/Stack';
import { CustomStackProps } from '../Stack/Stack';

const HStack: FCC<Omit<CustomStackProps, 'direction'>> = ({ spacing, className, children, ...props }) => {
  return (
    <Stack direction="row" spacing={spacing} className={className} {...props}>
      {children}
    </Stack>
  );
};

export default HStack;
