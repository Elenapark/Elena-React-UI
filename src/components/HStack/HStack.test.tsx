import { render, screen } from '@testing-library/react';
import React from 'react';
import { HStack } from './HStack';

const children = [<h2 id="test">hello</h2>, <h1>hi</h1>, 7, 'elenapark'];

const renderStack = () => {
  const defaultProps = {
    className: 'hstack',
    children,
  };
  render(<HStack {...defaultProps} />);
};

describe('<HStack />', () => {
  it('should only return row direction stack', () => {
    renderStack();
    expect(screen.getByTestId('stack-component')).toHaveAttribute('direction', 'row');
  });
});
