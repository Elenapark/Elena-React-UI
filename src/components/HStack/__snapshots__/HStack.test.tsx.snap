// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HStack /> should render 1`] = `
<div
  class="sc-bcXHqe eVfqAP stack hstack test"
  data-testid="stack-component"
  direction="row"
  spacing="8px"
>
  <h2
    data-testid="stack-child-comp"
    id="test"
  >
    hello
  </h2>
  <h1
    data-testid="stack-child-comp"
  >
    hi
  </h1>
  <div
    data-testid="stack-child-comp"
  >
    7
  </div>
  <div
    data-testid="stack-child-comp"
  >
    elenapark
  </div>
</div>
`;
